subsys/bluetooth/host/audio/chan.c

```diff
@@ -1569,6 +1564,40 @@ int bt_audio_broadcast_source_start(struct bt_audio_broadcast_source *source)
 	return 0;
 }
 
+int bt_audio_broadcast_source_stop(struct bt_audio_broadcast_source *source)
+{
+	struct bt_audio_chan *chan;
+	int err;
+
+	CHECKIF(source == NULL) {
+		BT_DBG("source is NULL");
+		return -EINVAL;
+	}
+
+	chan = source->chan;
+
+	if (chan->state != BT_AUDIO_CHAN_STREAMING) {
```

Comment:
For my education: Where/when during the proecess does the state change away from STREAMING?

Reply:
For a broadcast source, that is when the ISO disconnected callback is called, which is handled in endpoint.c:

```
static void ep_iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
{
	struct bt_audio_ep *ep = EP_ISO(chan);
	struct bt_audio_chan_ops *ops = ep->chan->ops;
	const bool is_broadcast = bt_audio_ep_is_broadcast(ep);

	BT_DBG("chan %p ep %p reason 0x%02x", chan, ep, reason);

	if (is_broadcast) {
		if (bt_audio_ep_is_broadcast_src(ep)) {
			bt_audio_chan_set_state(ep->chan,
						BT_AUDIO_CHAN_CONFIGURED);
	...
```